pipeline_utils/pipeline_deploy.py

```diff
@@ -0,0 +1,406 @@
+        os.mkdir(upload_)
+
+        # Read description files and create modified files for upload
+        #   placeholder variables will be replaced
+        #   with specific values for the target environment
+        files_ = glob.glob(f'{filepath_}/*.cwl')
+        files_.extend(glob.glob(f'{filepath_}/*.wdl'))
+        for fn in files_:
+            logger.info('> Processing %s' % fn)
+            # set file specific variables
+            file_ = f'{filepath_}/{fn}'
+            upload_file_ = f'{upload_}/{fn}'
+            s3_file_ = f'{self.pipeline}/{self.version}/{fn}'
+            if not self.debug:
+                # create modified description file for upload
+                with open(file_, 'r') as read_:
+                    with open(upload_file_, 'w') as write_:
+                        # replace generic variables
+                        for line in read_:
+                            line = line.replace('ACCOUNT', account_)
+                            line = line.replace('VERSION', self.version)
+                            line = line.replace('LICENSEID', self.sentieon_server)
+                            write_.write(line)
+                # upload to s3
+                extra_args = {'ACL': 'public-read'}  # note that this is no longer public if using encryption!
+                if self.kms_key_id:
+                    extra_args.update(update_)
+                s3.meta.client.upload_file(upload_file_, self.wfl_bucket, s3_file_, ExtraArgs=extra_args)
+                # delete file to allow tmp folder to be deleted at the end
+                os.remove(upload_file_)
+
+        # Clean tmp directory
+        os.rmdir(upload_)
+
+
+    def _post_patch_ecr(self, type='ECR'):
+        """
+        """
+        logger.info('@ Docker Image...')
+
+        # Set general variables
+        filepath_ = f'{self.repo}/{self.filepath[type]}'
+        account_ = f'{self.account}.dkr.ecr.{self.region}.amazonaws.com'
+
+        # Check
+        if not os.path.isdir(filepath_):
+            logger.error(f'WARNING: {self.filepath[type]} not found in {self.repo}, skipping...')
+            return
+
+        # Generic bash commands to be modified to correct version and account information
+        for fn in glob.glob(f'{filepath_}/*'):
+            logger.info('> Processing %s' % fn)
+            if not self.debug:
+                # set specific variables
+                tag_ = f'{account_}/{fn}:{self.version}'
+                path_ = f'{filepath_}/{fn}'
+                image = f"""
+                            aws ecr get-login-password --region {self.region} | docker login --username AWS --password-stdin {account_}
+                            docker build -t {tag_} {path_} --no-cache
+                            docker push {tag_}
+                        """ # note that we are ALWAYS doing no-cache builds so that we can get updated base images whenever applicable
+                subprocess.check_call(cmd, shell=True)
```

Comment:
I think this should be 
```suggestion
                subprocess.check_call(image, shell=True)
```

Reply:
right